service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala

```diff
@@ -56,46 +54,78 @@ class HttpRequestHandler(
       HttpVersion.HTTP_1_1,
       HttpResponseStatus.OK,
       Unpooled.copiedBuffer(response, CharsetUtil.UTF_8))
-    res.headers().set(HttpHeaderNames.CONTENT_TYPE, "text/html; charset=UTF-8");
-    ctx.writeAndFlush(res).addListener(ChannelFutureListener.CLOSE);
+    res.headers().set(HttpHeaderNames.CONTENT_TYPE, "text/html; charset=UTF-8")
+    ctx.writeAndFlush(res).addListener(ChannelFutureListener.CLOSE)
+  }
+
+  private def handleRequest(uri: String): String = {
+    if (service.serviceName == Service.MASTER) {
+      new MasterRequestHandler(service, uri).handle()
+    } else {
+      new WorkerRequestHandler(service, uri).handle()
+    }
   }
+}
+
+class BaseRequestHandler(service: HttpService, uri: String) extends Logging {
 
-  def handleRequest(uri: String): String = {
-    val (path, parameters) = HttpUtils.parseUrl(uri)
+  val (path, parameters) = HttpUtils.parseUrl(uri)
+
+  def handle(): String = {
     path match {
       case "/conf" =>
         service.getConf
-      case "/masterGroupInfo" if service.serviceName == Service.MASTER =>
-        service.getMasterGroupInfo
       case "/workerInfo" =>
         service.getWorkerInfo
-      case "/lostWorkers" if service.serviceName == Service.MASTER =>
-        service.getLostWorkers
-      case "/excludedWorkers" if service.serviceName == Service.MASTER =>
-        service.getExcludedWorkers
-      case "/shutdownWorkers" if service.serviceName == Service.MASTER =>
-        service.getShutdownWorkers
       case "/threadDump" =>
         service.getThreadDump
-      case "/hostnames" if service.serviceName == Service.MASTER =>
-        service.getHostnameList
-      case "/applications" if service.serviceName == Service.MASTER =>
-        service.getApplicationList
       case "/shuffles" =>
         service.getShuffleList
       case "/listTopDiskUsedApps" =>
         service.listTopDiskUseApps
-      case "/listPartitionLocationInfo" if service.serviceName == Service.WORKER =>
+      case _ => "invalid"
+    }
+  }
+}
+
+class MasterRequestHandler(service: HttpService, uri: String)
```

Comment:
@RexXiong, I wll raise another pull request to refactor `HttpRequestHandler`. BTW, does this need to introduce `/applicationCount`, `/partitionWritten` and `/partitionFileCount` Rest API in this handler?

Reply:
> @RexXiong, I wll raise another pull request to refactor `HttpRequestHandler`. BTW, does this need to introduce `/applicationCount`, `/partitionWritten` and `/partitionFileCount` Rest API in this handler?

No, these rest apis spec are too fine grained and duplicate since we can get from metrics, if we really want to get these info, we may use /shuffle/summary (or shuffleInfo for shot --- need discuss)for this.

Reply:
@RexXiong, I agree with the point that we could use ` /shuffle/summary` to provide users with shuffle statictics. I have   discussed with @FMX offline and he agreed that exposes the above Rest API. Anyway, I will expose this interface with the refactor in another pull request.